evalml/tests/component_tests/test_one_hot_encoder.py

```diff
@@ -441,3 +466,13 @@ def check_df_equality(random_state):
 
     check_df_equality(5)
     check_df_equality(get_random_state(5))
+
+
+def test_ohe_column_names_unique():
+    df = pd.DataFrame({"A": ["x_y"], "A_x": ["y"]})
+    df_transformed = OneHotEncoder().fit_transform(df)
+    assert set(df_transformed.columns) == {"A_x_y", "A_x_y_1"}
+
+    df = pd.DataFrame({"A": ["x_y", "z"], "A_x": ["y_1", "y"], "A_x_y": ["1", "y"]})
+    df_transformed = OneHotEncoder().fit_transform(df)
+    assert set(df_transformed.columns) == {"A_x_y", "A_z", "A_x_y_1", "A_x_y_1_1", "A_x_y_1_1_2", "A_x_y_y"}
```

Comment:
This looks kinda wild, although I guess this instance shouldn't occur very often for a user.

Reply:
Yea I don't think there is a perfect solution here in the sense that any solution will slightly obfuscate the column name/category level that results in each column in the transformed dataframe (you can create an adversarial example no matter the format we use to name the new columns).

That being said, I think it's important to have unique names to avoid any potential downstream bugs that can arise from having duplicate column names. 

Maybe we can better explain how the columns are named in the docs (and explain what happens when there are collisions) so users can better trace the data through the pipeline if they are debugging.